src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_add_nodepool_with_custom_ca_trust_certificates.yaml

```diff
@@ -0,0 +1,761 @@
+      CommandName:
+      - aks create
+      Connection:
+      - keep-alive
+      ParameterSetName:
+      - --resource-group --name --nodepool-name -c --ssh-key-value --aks-custom-headers
+        --custom-ca-trust-certificates
+      User-Agent:
+      - AZURECLI/2.70.0 (DOCKER) azsdk-python-core/1.31.0 Python/3.12.3 (Linux-6.8.0-1021-azure-x86_64-with-glibc2.38)
+    method: GET
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/clitest000001?api-version=2022-09-01
+  response:
+    body:
+      string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest000001","name":"clitest000001","type":"Microsoft.Resources/resourceGroups","location":"westus2","tags":{"product":"azurecli","cause":"automation","test":"test_aks_create_add_nodepool_with_custom_ca_trust_certificates","date":"2025-03-21T22:57:42Z","module":"acs"},"properties":{"provisioningState":"Succeeded"}}'
+    headers:
+      cache-control:
+      - no-cache
+      content-length:
+      - '392'
+      content-type:
+      - application/json; charset=utf-8
+      date:
+      - Fri, 21 Mar 2025 22:57:44 GMT
+      expires:
+      - '-1'
+      pragma:
+      - no-cache
+      strict-transport-security:
+      - max-age=31536000; includeSubDomains
+      x-cache:
+      - CONFIG_NOCACHE
+      x-content-type-options:
+      - nosniff
+      x-ms-ratelimit-remaining-subscription-global-reads:
+      - '16499'
+      x-msedge-ref:
+      - 'Ref A: F6F73CF690214017B16E9C71D6F4AE29 Ref B: BN1AA2051014033 Ref C: 2025-03-21T22:57:44Z'
+    status:
+      code: 200
+      message: OK
+- request:
+    body: '{"location": "westus2", "identity": {"type": "UserAssigned", "userAssignedIdentities":
+      {"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/e2e-managed-identity-pool/providers/Microsoft.ManagedIdentity/userAssignedIdentities/cp-268":
+      {}}}, "properties": {"kubernetesVersion": "", "dnsPrefix": "cliakstest-clitest22gn6sp7c-79a739",
+      "agentPoolProfiles": [{"count": 1, "vmSize": "Standard_DS2_v2", "osType": "Linux",
+      "enableAutoScaling": false, "type": "VirtualMachineScaleSets", "mode": "System",
+      "orchestratorVersion": "", "upgradeSettings": {}, "enableNodePublicIP": false,
+      "scaleSetPriority": "Regular", "scaleSetEvictionPolicy": "Delete", "spotMaxPrice":
+      -1.0, "nodeTaints": [], "enableEncryptionAtHost": false, "enableUltraSSD": false,
+      "enableFIPS": false, "name": "c000003"}], "linuxProfile": {"adminUsername":
+      "azureuser", "ssh": {"publicKeys": [{"keyData": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCL1KGP0wR222Ot/MECoGTmP+Owj+jxfcN8NCbI+FvQ7lYWGmUIgYMCzbLj0+aJAfcjLDT68yM9nEW4w9mK5i99P2va1jCnrh1l5UcstSm38x210xUJE7F0zJRAc8yZ1saYBfdEKZxuwm392AukQHXVxkF2WyZty3J/26m7xYuA7UkLCTAspLCegVO7rqf1mlfANnbFSsx/mq5daLtDEsfYJP49voLhIvrtOvG1iKREtUJ6mDWYtMdj3nZFwJlpPk/8zYsi+9RBNZTiYOpHuvWuVrZ3fHBWDgJAi2A8guDDHdmMJuNXQOuMQFg1jmZI7CB1W4k2ctq22ALPoQ3+upPH
```

Comment:
They may be scanned as potential secrets, so please replace them with fake ones (refer to this PR: https://github.com/Azure/azure-cli/pull/29951/files).

Reply:
Hi @yanzhudd , do you mean the SSH public key? Or the certificates from line 113?

Reply:
I modified the recording to remove the certificates, please let me know if this works @yanzhudd

Reply:
Actually this makes the CI checks fail, I reverted the commit for now. The certs passed in the input are dummy fake ones.
@FumingZhang  do you maybe know how this should be handled? AKS-RP validation requires valid certificates to be sent in input so if we modify the recording that we compare everything to, it would fail correct?

Reply:
Actually you can replace the value in recording files, in replay mode, the request won't be sent to ARM/AKS RP, so it's safe to put in any value you want there.

But the problem is, the property is defined with type bytearray and in that case, the value must be base64 encoded, even though I set the literal value to `testcert`, the encoded value `dGVzdGNlcnQ=` still looks like a cred and I’m not sure it could pass the test.

See example in latest commit [#972a888](https://github.com/FumingZhang/azure-cli/commit/972a888166998e6909fd6f438b3d84cca7112301) in branch [fuming/ca-fix-test-0325](https://github.com/FumingZhang/azure-cli/commits/fuming/ca-fix-test-0325/)

Reply:
Is it okay to leave the test certs as is in this case? Or should they be changed? Not sure what the best way to address this is @FumingZhang / @yanzhudd

Reply:
Hi @FumingZhang / @yanzhudd / @yonzhan I used Fuming's suggestion for test data formatting, could you please take another look and let me know if this works? Thanks!